src/views/ModelPlan/TaskList/Basics/index.tsx

```diff
@@ -1,29 +1,480 @@
+                              }) as string[]).map((item, key) => {
+                                return (
+                                  <Fragment key={item}>
+                                    <Field
+                                      as={CheckboxField}
+                                      id={`new-plan-cmmiGroup--${key}`}
+                                      name="cmmiGroup"
+                                      label={item}
+                                      value={translateCmmiGroup(item)}
+                                      checked={values.cmmiGroup.includes(
+                                        translateCmmiGroup(item)
+                                      )}
+                                      onChange={(
+                                        e: React.ChangeEvent<HTMLInputElement>
+                                      ) => {
+                                        if (e.target.checked) {
+                                          arrayHelpers.push(e.target.value);
+                                        } else {
+                                          const idx = values.cmmiGroup.indexOf(
+                                            e.target.value
+                                          );
+                                          arrayHelpers.remove(idx);
+                                        }
+                                      }}
+                                    />
+                                  </Fragment>
+                                );
+                              })}
+                            </>
+                          )}
+                        />
+                      </FieldGroup>
+                    )}
+
+                    <div className="margin-top-6 margin-bottom-3">
+                      <Button
+                        type="submit"
+                        disabled={!(dirty || isValid)}
+                        onClick={() => setErrors({})}
+                      >
+                        {h('next')}
+                      </Button>
+                    </div>
+                    <Button
+                      type="button"
+                      className="usa-button usa-button--unstyled"
+                      onClick={() => {
+                        if (values.modelName === '') {
+                          setFieldError('modelName', 'Enter the Model name');
+                        } else {
+                          handleFormSubmit(values, 'back');
+                        }
+                      }}
+                    >
+                      {/* TODO:
+                        2. Ensure all steps of form works
+                        3. milestone errors
+                      */}
+                      <IconArrowBack className="margin-right-1" aria-hidden />
+                      {h('saveAndReturn')}
+                    </Button>
+                  </Form>
+                  <AutoSave
```

Comment:
This autosave submithandler can behave as normal, except a conditional for if there is a modelName

```
<AutoSave
                    values={values}
                    onSave={() => {
                      if (formikRef.current!.values.modelName)
                        handleFormSubmit(formikRef.current!.values);
                    }}
                    debounceDelay={3000}
```